master/docs/manual/worker-transition.rst

```diff
@@ -0,0 +1,1091 @@
+     - :py:attr:`buildbot.locks.WorkerLock.maxCountForWorker`
+
+   * - :py:class:`buildbot.locks.SlaveLock` constructor argument
+       ``maxCountForSlave`` was renamed
+     - ``maxCountForWorker``
+
+
+   * - :py:mod:`buildbot.steps.slave`
+     - :py:mod:`buildbot.steps.worker`
+
+   * - :py:class:`buildbot.steps.slave.SlaveBuildStep`
+     - :py:class:`buildbot.steps.worker.WorkerBuildStep`
+
+   * - :py:class:`buildbot.steps.slave.CompositeStepMixin.getFileContentFromSlave`
+     - :py:class:`buildbot.steps.worker.CompositeStepMixin.getFileContentFromWorker`
+
+
+   * - :py:attr:`buildbot.steps.transfer.FileUpload.slavesrc`
+     - :py:attr:`~buildbot.steps.transfer.FileUpload.workersrc`
+
+   * - :py:class:`buildbot.steps.transfer.FileUpload`
+       constructor argument ``slavesrc`` was renamed to
+
+     - ``workersrc``
+
+
+   * - :py:attr:`buildbot.steps.transfer.DirectoryUpload.slavesrc`
+     - :py:attr:`~buildbot.steps.transfer.DirectoryUpload.workersrc`
+
+   * - :py:class:`buildbot.steps.transfer.DirectoryUpload`
+       constructor argument ``slavesrc`` was renamed to
+
+     - ``workersrc``
+
+
+   * - :py:attr:`buildbot.steps.transfer.MultipleFileUpload.slavesrcs`
+     - :py:attr:`~buildbot.steps.transfer.MultipleFileUpload.workersrcs`
+
+   * - :py:class:`buildbot.steps.transfer.MultipleFileUpload`
+       constructor argument ``slavesrcs`` was renamed to
+
+     - ``workersrcs``
+
+
+   * - :py:attr:`buildbot.steps.transfer.FileDownload.slavedest`
+     - :py:attr:`~buildbot.steps.transfer.FileDownload.workerdest`
+
+   * - :py:class:`buildbot.steps.transfer.FileDownload`
+       constructor argument ``slavedest`` was renamed to
+
+     - ``workerdest``
+
+
+   * - :py:attr:`buildbot.steps.transfer.StringDownload.slavedest`
+     - :py:attr:`~buildbot.steps.transfer.StringDownload.workerdest`
+
+   * - :py:class:`buildbot.steps.transfer.StringDownload`
+       constructor argument ``slavedest`` was renamed to
+
+     - ``workerdest``
+
+
+   * - :py:attr:`buildbot.steps.transfer.JSONStringDownload.slavedest`
+     - :py:attr:`~buildbot.steps.transfer.JSONStringDownload.workerdest`
+
+   * - :py:class:`buildbot.steps.transfer.JSONStringDownload`
+       constructor argument ``slavedest`` was renamed to
+
+     - ``workerdest``
+
+
+   * - :py:attr:`buildbot.steps.transfer.JSONPropertiesDownload.slavedest`
+     - :py:attr:`~buildbot.steps.transfer.JSONPropertiesDownload.workerdest`
+
+   * - :py:class:`buildbot.steps.transfer.JSONPropertiesDownload`
+       constructor argument ``slavedest`` was renamed to
+
+     - ``workerdest``
+
+   * - :py:attr:`buildbot.process.remotecommand.RemoteCommand.buildslave`
+     - :py:attr:`~buildbot.process.remotecommand.RemoteCommand.worker`
+
+
+API changes between 0.9.0b6 and 0.9.0b7 (done without providing fallback):
+
+.. todo::
+
+   This whole section may be removed since it's not important for users
+   upgrading to 0.9.0.
```

Comment:
Probably better to include it, with a note that users upgrading from 0.8.x to 0.9.0 don't need to worry about it (since we have a number of people using 0.9.0bX)